pkg/db/db.go

```diff
@@ -28,7 +29,7 @@ type DB struct {
 	BatchSize int
 }
 
-func New(dsn string) (*DB, error) {
+func New(dsn string, reportEnd *time.Time) (*DB, error) {
```

Comment:
I think it'd be cleaner if we separated the concerns of creating a DB client, and handling migrations in a Migrate() function.  You don't have to do it since this is a pre-existing thing but it would be nicer I think.

Reply:
I'm definitely open to some refactoring to consolidate / cleanup.  I could create a story about this and the utility methods for start, end and boundary we touched on in the previous pr that we closed without merging.  Would prefer to get this in an agreeable state first though then refactor after merging if ok.